include/aws/common/thread.h

```diff
@@ -57,17 +64,23 @@ AWS_COMMON_API int aws_thread_init(struct aws_thread *thread, struct aws_allocat
 /**
  * Creates an OS level thread and associates it with func. context will be passed to func when it is executed.
  * options will be applied to the thread if they are applicable for the platform.
+ * You must either call join or detach after creating the thread and before calling clean_up. 
  */
 AWS_COMMON_API int aws_thread_create(struct aws_thread *thread, void(*func)(void *arg),
```

Comment:
Potential point of confusion: I can see people not knowing that they need to call `_init` before `_create`. After all, shouldn't the create function initialize whatever it's creating?

I'd call it something different, like `_launch` instead. Also, you can avoid a lot of the compatibility/default issues with `aws_thread_options` by having setters on the thread structure instead.

Reply:
fair enough.